googletest/cmake/internal_utils.cmake

```diff
@@ -271,3 +271,32 @@ function(py_test name)
     endif (${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} GREATER 3.1)
   endif(PYTHONINTERP_FOUND)
 endfunction()
+
+include(CMakePackageConfigHelpers)
+
+# create_cmake_package(package_name)
+#
+# Create a CMake package to enable 'find_package' work in 'config' mode.
+function(create_cmake_package package_name)
+  set(generated_dir "${CMAKE_CURRENT_BINARY_DIR}/generated")
+  set(install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${package_name}")
+  set(targets_export_name ${package_name}Targets)
+  # Export the CMake targets.
+  install(
+    EXPORT ${targets_export_name}
+    NAMESPACE ${package_name}::
+    DESTINATION ${install_dir})
+  # Export the package version.
+  set(version_file "${generated_dir}/${package_name}ConfigVersion.cmake")
+  write_basic_package_version_file(${version_file} COMPATIBILITY AnyNewerVersion)
+  # Wrap the exported targets in the CMake package utility.
+  set(config_file "${generated_dir}/${package_name}Config.cmake")
+  configure_package_config_file(
+    "${CMAKE_SOURCE_DIR}/googletest/cmake/Config.cmake.in"
```

Comment:
Variable `CMAKE_SOURCE_DIR` is a top-level directory of source tree. This code will not work correctly if GTest will be added by `add_subdirectory`. You can use `${gtest_SOURCE_DIR}/googletest/cmake/Config.cmake.in` instead.

Reply:
Agreed. Fixed.